### ISSUE TITLE
Hole in the middle of a record is malformed

### ISSUE THREAD

**Issue by laMudri** [agda/agda#5463]
The following code gives the error below.

Code:

```agda
open import Agda.Primitive using (Level; lsuc)

record Foo (o : Level) : Set (lsuc o) where

  field
    Obj : Set o
    Hom : Set o

  Sub : Set ?
  Sub = Hom

  field
    id : Set o
```

Error:

```
Set o != Set (λ Hom₁ → o)
when checking that the expression Hom has type Set (λ Hom₁ → o)
```

**Comment by andreasabel** [comment]
Similar: #5478, #2257

**Comment by andreasabel** [comment]
After #5483 the error is now:
```
Set o != Set Hom
```

**Comment by andreasabel** [comment]
Seems like this was fixed by my work on #5478.
